### DISCUSSION TITLE
How to configure connection config so that it will accept only IKE_COMPLIANT certificates.

### DISCUSSION THREAD

**Discussion by digambaringale333** [strongswan/strongswan#1100] [Q&A] [answered]
Hi,
We wanted to make a connection config which should accept only IKE_COMPLIANT certificate to create IPSec tunnels.
We searched for it and found that there is parameter in https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html, cert_policy where we can specify comma-separated list of certificate policy OIDs the peer’s certificate must have. 

IKE compliance in strongswan code:
```
        /** cert either lacks keyUsage bits, or includes either digitalSignature
         *  or nonRepudiation as per RFC 4945, section 5.1.3.2. */
        X509_IKE_COMPLIANT =      (1<<10),

```
RFC  : https://datatracker.ietf.org/doc/html/rfc4945#section-5.1.3.2
Here it say's we should have either digital signature OR non repudiation flag set in certificates.

How we can enforce this compliance in connection config? if we can use cert_policy for the same then what OID we should mention in cert_policy to enforce this compliance ?

Any help on this much appreciated.

**Comment by tobiasbrunner** [comment, marked as answer]
> How we can enforce this compliance in connection config? 

It's enforced since 5.6.3 (there is no option to disable it).

**Reply by digambaringale333** [reply]
Thanks Tobias!
